### PR TITLE
feat(core) Add Mouse capture support, set/get mouse pos, set custom cursor image

### PR DESCRIPTION
**Pull request by hironichu**
Hello,

I wanted to use DWM few days ago and found out that few methods were not yet implemented with GLFW,

This PR is adding some specific bindings that I needed for my prototype : 

-  glfwSetInputMode
-  glfwGetInputMode
-  glfwRawMouseMotionSupported
-  glfwSetCursorPos

As a "bonus" i tried implementing `setCustomCursor` using the new FFI structs, tho I am very new to this so it might be something to re-write (it should work tho?)

Putting this PR in draft for now.

### CODE REVIEW

**Comment by load1n9** [comment]
looking good so far

File: src/platform/glfw/window.ts [hunk 1]
```diff
@@ -847,7 +869,25 @@ export class WindowGlfw extends DwmWindow {
       }
     }
   }
-
+  setCustomCursor(image: Uint8Array, hotspot: Size, position: Position) {
```
Comment (load1n9) [inline src/platform/glfw/window.ts:872]: once the new byte type gets released lets create the struct that way for now this works

**Comment by load1n9** [comment]
maybe also make an example for this

File: src/platform/glfw/window.ts [hunk 2]
```diff
@@ -847,7 +869,25 @@ export class WindowGlfw extends DwmWindow {
...
+    const stringptr = Deno.UnsafePointer.of(image);
+    const struct = new Uint8Array(16);
+    const structview = new DataView(struct.buffer);
+    structview.setInt32(0, hotspot.width, true);
```
Comment (DjDeveloperr) [inline src/platform/glfw/window.ts:876]: nit: Assumes native endianness to be LE. Not sure if it matters for now though.

**Comment by DjDeveloperr** [comment]
LGTM. Is it tested? Can you add an example in `examples/` to demonstrate the API?

**Comment by hironichu** [comment]
I have tested this myself and it works (on mac and windows as far as I know), will add test later today.

Quick note on the Custom cursor image, I've tested this while using a buffer with valid RGBA pixels data, however reading directly from PNG doesn't  work, (it doesn't break its just messy pixels with wrong colors like invalid texture in Gary's Mod).. like I said it might need more work, i will try to add examples for this as well but we can also remove it and do a better implementation later.

**Comment by hironichu** [comment]
I made two examples for the methods that really needed it, there are other stuff we can do with `glfwSetInputMode` but going into detail can be done by just reading Glfw doc.

**Review by DjDeveloperr — APPROVED** [review]
LGTM, thanks!

**Pull request merged**